pkg/controllers/automigration/controller.go

```diff
@@ -301,40 +301,64 @@ func (c *Controller) estimateCapacity(
 	estimatedCapacity := make(map[string]int64, len(clusterObjs))
 
 	for _, clusterObj := range clusterObjs {
-		unsObj := clusterObj.Object.(*unstructured.Unstructured)
+		keyedLogger := keyedLogger.WithValues("cluster", clusterObj.ClusterName)
```

Comment:
Do we want to insert the new logger into the context? We should probably provide a helper for this in case we forget.